pkg/proof/proof.go

```diff
@@ -68,25 +65,30 @@ func NewShareInclusionProof(
 
 	eds, err := da.ExtendShares(shares.ToBytes(dataSquare))
 	if err != nil {
-		return types.ShareProof{}, err
+		return ShareProof{}, err
 	}
 
 	edsRowRoots, err := eds.RowRoots()
 	if err != nil {
-		return types.ShareProof{}, err
+		return ShareProof{}, err
 	}
 
 	edsColRoots, err := eds.ColRoots()
 	if err != nil {
-		return types.ShareProof{}, err
+		return ShareProof{}, err
 	}
 
 	// create the binary merkle inclusion proof for all the square rows to the data root
 	_, allProofs := merkle.ProofsFromByteSlices(append(edsRowRoots, edsColRoots...))
-	rowProofs := make([]*merkle.Proof, endRow-startRow+1)
-	rowRoots := make([]tmbytes.HexBytes, endRow-startRow+1)
+	rowProofs := make([]*Proof, endRow-startRow+1)
+	rowRoots := make([][]byte, endRow-startRow+1)
 	for i := startRow; i <= endRow; i++ {
-		rowProofs[i-startRow] = allProofs[i]
+		rowProofs[i-startRow] = &Proof{
+			Total:    allProofs[i].Total,
+			Index:    allProofs[i].Index,
+			LeafHash: allProofs[i].LeafHash,
+			Aunts:    allProofs[i].Aunts,
+		}
```

Comment:
Yeah because there are two different types (one from go-square and one from celestia-app because we use gogoproto in app and don't in go-square so it generates the structs with different methods).

I have an idea of how we can resolve this in the future